packages/tpetra/core/test/MatrixMatrix/MatrixMatrix_UnitTests.cpp

```diff
@@ -1019,20 +1020,37 @@ TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL(Tpetra_MatMat, operations_test,SC,LO, GO, NT)
       newOut << "\tComputed norm: " << results.computedNorm << endl;
       newOut << "\tEpsilon: " << results.epsilon << endl;
 
-      B = Reader<Matrix_t >::readSparseFile(B_file, comm, false);
+      B = Reader<Matrix_t >::readSparseFile(B_file, comm, true, false, false);
+      //declare D with enough entries to receive A + B
+      size_t n = 0;
+      for (size_t i = 0; i < B->getGlobalNumRows(); i++) {
+	if (n < B->getNumEntriesInGlobalRow(i))
+	  n = B->getNumEntriesInGlobalRow(i);
+      }
+      n += A->getGlobalMaxNumRowEntries();
+
+      newOut << "construct row map" << endl;
+      RCP<const map_type> rm = B->getRowMap();
+      newOut << "construct new Sparse E" << endl;
+      RCP<Matrix_t> E = rcp (new Matrix_t(rm, n, Tpetra::StaticProfile));
+      newOut << "get one" << endl;
+      auto one = Teuchos::ScalarTraits<SC>::one();
+      newOut << "add B to new sparse matrix" << endl;
+      Tpetra::MatrixMatrix::Add(*B, BT, one, *E, one);
```

Comment:
We might actually need to fix Add to do the right thing here.

Reply:
I don't see a way for Add to work on non-DynamicProfiles without something like this, though maybe there's a more elegant way to do it inside MatrixMatrix than in the unit test.

Honestly, I'm surprised that more didn't break within Tpetra that got caught by the testing.

Suggestions on an approach to fix?

Reply:
The logical thing to do is to make Add do the same thing that Export / Import to a StaticProfile CrsGraph does (thanks to @tjfulle ), namely reallocate if needed.

Reply:
In this case "do the right thing" according to the docs is to make B big enough on input to the += to allow it to hold the whole input.

I'll look into this on Friday to see how to do that.

Reply:
So, it appears that I've made B big enough on input to the += to allow it to hold the whole input.

This way, we keep the test, and show a possible way to work around the issue.